test/controllers/charge_controller_create_test.js

```diff
@@ -0,0 +1,199 @@
+    'corporate_debit_card_surcharge_amount': 0
+  }
+}
+
+const mockedChargeValidationBackend = function () {
+  const validation = {
+    hasError: false
+  }
+  return {
+    verify: () => {
+      return Promise.resolve({ validation, card })
+    }
+  }
+}
+
+const requireChargeController = function (mockedConnectorClient) {
+  const proxyquireMocks = {
+    '../utils/charge_validation_backend': mockedChargeValidationBackend,
+    '../services/clients/connector_client': mockedConnectorClient
+  }
+  return proxyquire(path.join(__dirname, '/../../app/controllers/charge_controller.js'), proxyquireMocks)
+}
+
+describe('POST /card_details/{chargeId} endpoint', function () {
+  let response
+  let chargeAuthStub
+  let mockedConnectorClient
+  let paymentDetails
+
+  beforeEach(() => {
+    chargeAuthStub = sinon.stub().resolves(
+      {
+        statusCode: 200,
+        body: {
+          status: 'AUTHORISATION SUCCESS'
+        }
+      })
+    mockedConnectorClient = sinon.stub().callsFake(() => {
+      return {
+        chargeAuth: chargeAuthStub
+      }
+    })
+    response = {
+      redirect: sinon.spy(),
+      locals: {
+        service: {
+          collectBillingAddress: true
+        }
+
+      }
+    }
+    paymentDetails = {
+      'chargeId': chargeId,
+      'cardNo': '4242424242424242',
+      'expiryMonth': '01',
+      'expiryYear': '20',
+      'cardholderName': 'Joe Bloggs',
+      'cvc': '111',
+      'addressCountry': 'GB',
+      'addressLine1': '1 Horse Guards',
+      'addressCity': 'London',
+      'addressPostcode': 'E1 8QS'
+    }
+  })
+
+  it('should send worldpay_3ds_flex_ddc_result to connector when the request includes a worldpay3dsFlexDdcResult parameter', async function () {
+    paymentDetails['worldpay3dsFlexDdcResult'] = 'a-worldpay-3ds-flex-ddc-result'
+    const request = {
+      chargeData: chargeData,
+      body: paymentDetails,
+      chargeId: chargeId,
+      header: sinon.spy(),
+      headers: {
+        'x-request-id': 'unique-id'
+      }
+    }
+    await requireChargeController(mockedConnectorClient).create(request, response)
+    expect(chargeAuthStub.calledWith(sinon.match( // eslint-disable-line
+      {
+        'chargeId': chargeId,
+        'payload': {
+          'card_number': paymentDetails.cardNo,
+          'cvc': paymentDetails.cvc,
+          'card_brand': card.brand,
+          'expiry_date': `${paymentDetails.expiryMonth}/${paymentDetails.expiryYear}`,
+          'cardholder_name': paymentDetails.cardholderName,
+          'card_type': card.type,
+          'corporate_card': card.corporate,
+          'prepaid': card.prepaid,
+          'address': {
+            'line1': paymentDetails.addressLine1,
+            'city': paymentDetails.addressCity,
+            'postcode': paymentDetails.addressPostcode,
+            'country': paymentDetails.addressCountry
+          },
+          'worldpay_3ds_flex_ddc_result': paymentDetails.worldpay3dsFlexDdcResult
+        }
```

Comment:
I think we should use `payment_fixtures`#`validAuthorisationRequest` to construct this JSON.

Reply:
I've done this but it's increased the number of lines of code as I've had to delete the accept_header and user_agent_header from the construct returned by `validAuthorisationRequest`. This is necessary as I've mocked out the calls to `req.header` with `header: sinon.spy()` earlier on in the test.